### PR TITLE
Feature/group picture upload

### PR DESCRIPTION
**Pull request by NChitty**
Need code review on image model and upload controller in controllers/group.js

Added documentation for endpoint.

### CODE REVIEW

File: src/controllers/group.js [hunk 1]
```diff
@@ -112,6 +115,53 @@ const Group = {
...
+    }
+
+    const image = new ImageModel(
+      fileName, userId, new Date(),
+    );
+
+    result = await GroupModel.update(
+      { _id: result.id },
+      { $push: { images: image } },
+    );
```
Comment (suneettipirneni) [inline src/controllers/group.js:161]: You can use `GroupModel.findByIDandUpdate()` here instead as it's more concise.

File: src/models/image.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const modelName = 'Image';
+
+const imageSchema = new mongoose.Schema({
+  fileName: { type: String, required: true, unique: true },
+  creator: { type: mongoose.Schema.Types.ObjectId, ref: User },
+  dateUploaded: { type: Date, required: true },
+});
+
+const populateFields = 'fileName dateUploaded';
```
Comment (suneettipirneni) [inline src/models/image.js:12]: I don't think `filename` needs to be populated, it's just a string

**Review by suneettipirneni — CHANGES_REQUESTED** [review]
Looks pretty good so far, just some things I spotted

File: src/schemas.yml [hunk 3]
```diff
@@ -103,3 +107,25 @@ GroupRequest:
...
+GroupUpload:
+  type: object
+  properties:
+    body:
+      type: object
+      format: json
+      $ref: '#/components/schemas/Group'
+    picture:
+      type: string
+      format: binary
```
Comment (Astatium5) [inline src/schemas.yml:119]: I would've added an example

File: src/schemas.yml [hunk 4]
```diff
@@ -103,3 +107,25 @@ GroupRequest:
...
+Image:
+  description: An Image object
+  type: object
+  properties:
+    fileName:
+      type: string
+      required: true
```
Comment (Astatium5) [inline src/schemas.yml:126]: Example here as well

File: src/schemas.yml [hunk 5]
```diff
@@ -103,3 +107,25 @@ GroupRequest:
...
+    creator:
+      type: string
+    dateUploaded:
+      type: string
+      format: date
```
Comment (Astatium5) [inline src/schemas.yml:131]: And here, as well as an empty string at the end of the file

File: src/routes/group.js [hunk 6]
```diff
@@ -163,4 +165,45 @@ groups.delete('/:id', Group.delete);
...
+ *     tags:
+ *     - Group
+ *
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         required: true
+ *         schema:
+ *          type: string
+ *         description: The ID of the group to upload the picture to
```
Comment (Astatium5) [inline src/routes/group.js:184]: Looks like this is invalid in terms of tabs

**Review by Astatium5 — COMMENTED** [review]
All in all good, just some minor changes

File: src/controllers/group.js [hunk 7]
```diff
@@ -112,6 +115,53 @@ const Group = {
...
+
+    return res.status(200).send(result.toJSON());
```
Comment (Armintron) [inline src/controllers/group.js:158]: `result` is a query object, which does not have a `.toJSON()` method.

**Review by Armintron — CHANGES_REQUESTED** [review]
Great work! Some simple mistakes cause the endpoint to fail, but they are easy fixes.

File: src/controllers/group.js (hunk 7 above)
Comment (suneettipirneni) [reply src/controllers/group.js:161]: I don't think we need to return any data to the response I would just have 204 response

Comment (suneettipirneni) [reply src/controllers/group.js]: I went ahead an made the change

**Review by suneettipirneni — APPROVED** [review]

**Pull request merged**